<a href="https://colab.research.google.com/github/somas1/CT/blob/main/NBA_PANDAS/NBA_Regular_Season_2018_19_Data_Challenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Nba Regular Season 2018-19 Data Challenge

Your task will be to take the dataset given, and create an analysis answering the following 10 questions. This project will again test your knowledge of pandas in order to find the answers needed given the data you are presented with.

In [2]:
import pandas as pd
import numpy as np

In [3]:
nba = pd.read_csv('nbastats2018-2019.csv')
nba.head()

Name  Height  Weight                   Team  Age    Salary  Points  \
0  Alex Abrines      78     200  Oklahoma City Thunder   25   5455236     5.3   
1    Quincy Acy      79     240           Phoenix Suns   28    213949     1.7   
2  Jaylen Adams      74     190          Atlanta Hawks   22    236854     3.2   
3  Steven Adams      84     265  Oklahoma City Thunder   25  24157304    13.9   
4   Bam Adebayo      82     255             Miami Heat   21   2955840     8.9   

   Blocks  Steals  Assists  ...    MP   G   PER  OWS  DWS   WS   WS48   USG  \
0     0.2     0.5      0.6  ...  19.0  31   6.3  0.1  0.6  0.6  0.053  12.2   
1     0.4     0.1      0.8  ...  12.3  10   2.9 -0.1  0.0 -0.1 -0.022   9.2   
2     0.1     0.4      1.9  ...  12.6  34   7.6 -0.1  0.2  0.1  0.011  13.5   
3     1.0     1.5      1.6  ...  33.4  80  18.5  5.1  4.0  9.1  0.163  16.4   
4     0.8     0.9      2.2  ...  23.3  82  17.9  3.4  3.4  6.8  0.171  15.8   

   BPM  VORP  
0 -3.4  -0.2  
1 -5.9  -0.1  
2 -4.4  -0.3  
3  2.7   3.2  
4  3.0   2.4  

[5 rows x 27 columns]

# What was the average age of player in the league?

In [8]:
nba['Age'].mean().round(2)

25.9

# What player scored the most points?

In [39]:
nba.sort_values('Points', ascending=False)[['Name', 'Points']].head(1)

Name  Points
202  James Harden    36.1

James Harden had a higher point value than any other player in the dataset at 36.1. This suggests the values in the dataset are averages for the season, not totals.

Since the values presented are most likely averages, I will avoid applying functions such as .mean() to the dataset to average values further.

# What player had the most blocks during the season? Was it a post player (F/C)?

In [56]:
most_blocks = nba.sort_values(by='Blocks', ascending=False)[['Name', 'Height', 'Points', 'Blocks', 'Rebounds', 'Assists', 'FGA']].head(1)
most_blocks

Name  Height  Points  Blocks  Rebounds  Assists   FGA
476  Myles Turner      83    13.3     2.7       7.2      1.6  10.5

In [69]:
shot_attempts = nba
shot_attempts

Height  Weight        Age  Points  Blocks    Steals    Assists  \
FGA                                                                       
24.5  77.000000   220.0  29.000000   36.10    0.70  2.000000   7.500000   
21.0  81.000000   220.0  28.000000   28.00    0.40  2.200000   4.100000   
20.5  73.000000   184.0  28.000000   25.60    0.40  1.200000   5.900000   
20.2  75.000000   200.0  30.000000   22.90    0.50  1.900000  10.700000   
19.9  77.500000   232.5  28.000000   25.60    0.50  1.350000   6.250000   
...         ...     ...        ...     ...     ...       ...        ...   
1.0   78.000000   206.2  23.600000    1.74    0.06  0.000000   0.400000   
0.9   81.000000   238.0  24.000000    0.80    0.00  0.000000   0.100000   
0.8   77.000000   215.0  24.000000    1.00    0.10  0.200000   0.300000   
0.7   80.666667   215.0  23.666667    0.30    0.00  0.033333   0.066667   
0.0   77.500000   195.5  23.000000    0.00    0.00  0.000000   0.000000   

       Rebounds     FT%        FTA  ...         MP          G    PER  \
FGA                                 ...                                
24.5   6.600000  0.8790  11.000000  ...  36.800000  78.000000  30.60   
21.0   8.200000  0.8390   7.000000  ...  36.900000  77.000000  23.30   
20.5   4.400000  0.8440   5.500000  ...  34.900000  82.000000  21.70   
20.2  11.100000  0.6560   6.200000  ...  36.000000  73.000000  21.10   
19.9   6.300000  0.7355   6.350000  ...  34.450000  66.000000  21.40   
...         ...     ...        ...  ...        ...        ...    ...   
1.0    0.860000  1.0000   0.400000  ...   3.000000   1.800000  31.80   
0.9    0.700000  1.0000   0.200000  ...   3.500000  11.000000   5.00   
0.8    0.700000  0.7350   0.500000  ...   4.900000  64.000000   5.20   
0.7    0.466667  0.3330   0.033333  ...   2.166667  10.333333 -11.10   
0.0    0.500000     NaN   0.000000  ...   4.000000   1.000000   0.95   

            OWS       DWS         WS      WS48    USG    BPM  VORP  
FGA                                                                 
24.5  11.400000  3.800000  15.200000  0.254000  40.50  11.70   9.9  
21.0   7.000000  4.900000  11.900000  0.201000  29.50   5.50   5.3  
20.5   5.500000  1.900000   7.400000  0.123000  31.50   3.30   3.9  
20.2   1.800000  5.000000   6.800000  0.124000  30.90   6.50   5.6  
19.9   3.000000  3.150000   6.100000  0.135500  31.60   4.35   3.3  
...         ...       ...        ...       ...    ...    ...   ...  
1.0    0.000000  0.000000   0.000000  0.347800  23.42   2.18   0.0  
0.9    0.000000  0.000000   0.000000  0.026000  13.00  -6.90   0.0  
0.8   -0.300000  0.300000   0.000000  0.004000  11.30  -4.00  -0.2  
0.7    0.033333  0.033333   0.066667 -0.370333  21.80 -20.20   0.0  
0.0    0.000000  0.000000   0.000000 -0.017000   0.00  -8.95   0.0  

[165 rows x 23 columns]

Myles Turner had the most blocks during the 2018-2019 season.

A post player plays with their back to the basket, which gives them the ability to block more shots than they can typically make.

He is 83 inches or 6'9 inches tall which suggests he's plays center. Center players are usually post players.

Someone with a high number of FGAs is likely to be a guard.

# Based on the regular season, who had the best chance to win a title given their win percentage?

# What player had the best 3-pt percentage?

# Who played the most minutes during the season

# What player given their player effiecency rating was the clutchest during the season?

# What team had the youngest roster?

# Who is the highest paid player during the seasion?

# At the end of a game, who WOULDN'T you want on the Free Throw Line?